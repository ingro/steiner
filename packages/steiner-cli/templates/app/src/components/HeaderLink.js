import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Route, Link } from 'react-router-dom';

export default class HeaderLink extends Component {
    render() {
        const { to, text } = this.props;

        return (
            <Route
                path={to}
                children={({ match }) => {
                    return (
                        <li className={match ? 'active' : ''}>
                            <Link to={to}>{text}</Link>
                        </li>
                    );
                }}
            />
        );
    }
}

HeaderLink.propTypes = {
    text: PropTypes.string.isRequired,
    to: PropTypes.string.isRequired
};