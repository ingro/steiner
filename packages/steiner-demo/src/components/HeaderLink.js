import React, { PropTypes } from 'react';
import { Link } from 'react-router';

const HeaderLink = (props) => <Link to={props.to} activeClassName="active" location={props.location}>{
    ({ isActive, onClick, href }) => <li className={isActive ? 'active' : ''}>
        <a href={href} onClick={onClick}>{props.name}</a>
    </li>
}</Link>;

HeaderLink.propTypes = {
    to: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired
};

export default HeaderLink;