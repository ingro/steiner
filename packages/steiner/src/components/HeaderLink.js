import React, { PropTypes } from 'react';
import { Link } from 'react-router-dom';

const HeaderLink = (props) => <Link to={props.to} location={props.location}>{
    ({ isActive, onClick, href }) => <li className={isActive ? 'active' : ''}>
        <a href={href} onClick={onClick}>{props.name}</a>
    </li>
}</Link>;

HeaderLink.propTypes = {
    to: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired
};

export default HeaderLink;