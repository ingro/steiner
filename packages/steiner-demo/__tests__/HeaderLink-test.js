import renderer from 'react-test-renderer';
import React from 'react';

import HeaderLink from '../src/HeaderLink';

test('HeaderLink renders correctly', () => {
    const tree = renderer.create(
        <HeaderLink to="/foo" name="Foo" location={{ pathname: '/foo/bar' }} />
    ).toJSON();

    expect(tree).toMatchSnapshot();
});