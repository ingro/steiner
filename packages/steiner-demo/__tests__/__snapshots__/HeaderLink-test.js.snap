exports[`test HeaderLink renders correctly 1`] = `
<li
  className="active">
  <a
    href="/foo"
    onClick={[Function anonymous]}>
    Foo
  </a>
</li>
`;
